backend/chunker/chunker_test.go

```diff
@@ -0,0 +1,194 @@
+// Test the Chunker filesystem interface
+package chunker_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	_ "github.com/rclone/rclone/backend/all" // for integration tests
+	"github.com/rclone/rclone/backend/chunker"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/fserrors"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/fs/object"
+	"github.com/rclone/rclone/fs/operations"
+	"github.com/rclone/rclone/fstest"
+	"github.com/rclone/rclone/fstest/fstests"
+	"github.com/rclone/rclone/lib/readers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Command line flags
+var (
+	// Invalid characters are not supported by some remotes, eg. Mailru.
+	// We enable testing with invalid characters when -remote is not set, so
+	// chunker overlays a local directory, but invalid characters are disabled
+	// by default when -remote is set, eg. when test_all runs backend tests.
+	// You can still test with invalid characters using the below flag.
+	TestBadChars = flag.Bool("bad-chars", false, "Set to test bad characters in file names when -remote is set")
+	UploadSizeMB = flag.Int("upload-megabytes", 0, "Upload size in megabytes")
+)
+
+// TestIntegration runs integration tests against a concrete remote
+// set by the -remote flag.
+func TestIntegration(t *testing.T) {
+	if *fstest.RemoteName == "" {
+		t.Skip("Skipping as -remote not set")
+		return
+	}
+	if *UploadSizeMB > 0 {
+		t.Skip("Skipping as -upload-megabytes is set")
+		return
+	}
+	fstests.Run(t, &fstests.Opt{
+		RemoteName:                   *fstest.RemoteName,
+		NilObject:                    (*chunker.Object)(nil),
+		UnimplementableFsMethods:     []string{"OpenWriterAt"},
+		UnimplementableObjectMethods: []string{"MimeType"},
+		SkipBadWindowsCharacters:     !*TestBadChars,
+	})
+}
+
+// TestStandard runs standard integration tests against chunker overlay
+// over a local tempdir, runs when the -remote flag is not set.
+func TestStandard(t *testing.T) {
+	if *fstest.RemoteName != "" {
+		t.Skip("Skipping as -remote is set")
+	}
+	tempdir := filepath.Join(os.TempDir(), "rclone-chunker-test-standard")
+	name := "TestChunker"
+	fstests.Run(t, &fstests.Opt{
+		RemoteName: name + ":",
+		NilObject:  (*chunker.Object)(nil),
+		ExtraConfig: []fstests.ExtraConfigItem{
+			{Name: name, Key: "type", Value: "chunker"},
+			{Name: name, Key: "remote", Value: tempdir},
+		},
+		UnimplementableFsMethods:     []string{"OpenWriterAt"},
+		UnimplementableObjectMethods: []string{"MimeType"},
+	})
+}
+
+// TestLargeUpload tests large file upload. It runs when
+// both -remote and -upload-megabytes flags are set.
+func TestLargeUpload(t *testing.T) {
```

Comment:
Note that you want to implement `fstests.InternalTester` by creating a function called `func (f *Fs) InternalTest(t *testing.T)` then this can be called as part of the integration tests so it doesn't have to do all it's own setup.

See the drive tests for an example

https://github.com/rclone/rclone/blob/e2ce687f9377813039a1614658ef2d1952de7a90/backend/drive/drive_internal_test.go#L271-L287